src/test/java/com/endava/internship/mocking/service/PaymentServiceTest.java

```diff
@@ -1,31 +1,131 @@
 package com.endava.internship.mocking.service;
 
+import com.endava.internship.mocking.model.Payment;
+import com.endava.internship.mocking.model.Status;
+import com.endava.internship.mocking.model.User;
+import com.endava.internship.mocking.repository.InMemPaymentRepository;
+import com.endava.internship.mocking.repository.InMemUserRepository;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.function.Executable;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import static org.assertj.core.api.Assertions.assertThat;
 @ExtendWith(MockitoExtension.class)
 class PaymentServiceTest {
 
-
+    @Mock
+    private InMemUserRepository userRepository;
+    @Mock
+    private InMemPaymentRepository paymentRepository;
+    @Mock
+    private BasicValidationService validationService;
+    @InjectMocks
+    private PaymentService paymentService =
+            new PaymentService(userRepository, paymentRepository, validationService);
     @BeforeEach
     void setUp() {
 
     }
 
     @Test
     void createPayment() {
+        // Given
+        double amount = 1D;
+        User user = new User(1, "John", Status.ACTIVE);
+        Payment expectedPayment = new Payment(user.getId(), amount, "Payment from user John");
+        ArgumentCaptor<Payment> paymentCaptor = ArgumentCaptor.forClass(Payment.class);
+
+        // When
+        when(userRepository.findById(user.getId())).thenReturn(Optional.of(user));
+        when(paymentRepository.save(any())).thenReturn(expectedPayment);
+        Payment result = paymentService.createPayment(user.getId(), amount);
+        verify(paymentRepository).save(paymentCaptor.capture());
 
+        // Then
```

Comment:
It's good that you're using Captor, nice job

Reply:
Thanks!